pkg/loader/load.go

```diff
@@ -229,6 +229,25 @@ func (s *loaderImpl) getTableInfo(schema string, table string) (info *tableInfo,
 	return s.refreshTableInfo(schema, table)
 }
 
+func needRefreshTableInfo(sql string) bool {
+	stmt, err := parser.New().ParseOneStmt(sql, "", "")
+	if err != nil {
+		log.Error("parse sql failed", zap.String("sql", sql), zap.Error(err))
+		return false
+	}
+
+	switch stmt.(type) {
+	case *ast.DropTableStmt:
+		return false
+	case *ast.DropDatabaseStmt:
+		return false
+	case *ast.TruncateTableStmt:
+		return false
```

Comment:
How about merging them in one case? It seems like we are doing that in other places like `IgnoreDDLError`.

Reply:
Why is this marked "resolved"? There are still 4 cases which can be merged into one.